components/sup/src/manager/service/mod.rs

```diff
@@ -732,11 +706,6 @@ impl Service {
         Ok(())
     }
 
-    fn run_hook(&mut self, hook: HookType) -> Result<()> {
-        self.hooks.compile(&self.service_group, &self.config);
-        self.hooks.try_run(hook, &self.service_group)
-    }
-
     /// Update our own service rumor with a new configuration from the packages exported
```

Comment:
funny @reset, my hook compile PR earlier initially removed the `run_hook` function from `service/mod` and replaced all `run_hook` calls with `self.hooks.try_run` and I reverted it because I was nervous that handlebar templating may have changed data that needs to be recompiled prior to running the hook. Like for example, a member is confirmed dead changing the live members, if you do not recompile the templates on every run, you risk having stale members in your hook no?